src/rust/engine/src/nodes/mod.rs

```diff
@@ -254,6 +262,13 @@ impl NodeKey {
         }
     }
 
+    pub fn fs_path_to_watch(&self) -> Option<&Path> {
+        match self {
+            NodeKey::PathMetadata(fs) => Some(fs.path().parent().unwrap_or_else(|| Path::new("."))),
+            _ => self.fs_subject(),
+        }
+    }
+
```

Comment:
This method allows there to be a difference between the path on which to place the filesystem watch and the path which actually invalidates this node. This is necessary because in the existing logic, if the path does not exist, then trying to place a watch on the path will fail with a "file not found" error.

The solution is to place the watch on the parent directory but only react when a creation / modification event is seen for the subject path.

Reply:
Could you add a comment on the `PathMetadata` explaining this background so we don't forget it? Thanks

Reply:
Will do.

Reply:
Done.